src/utils.jl

```diff
@@ -81,19 +81,18 @@ end
 On Linux systems, return the strings returned by the `uname()` function in libc.
 """
 function uname()
-    if !Sys.islinux()
+    @static if !Sys.islinux()
         return String[]
     end
 
     # Get libc (or musl) and handle to uname
-    possible_libc_names = String["libc.so"]
-    possible_musl_names = String["ld-musl-x86_64.so"]
-    libcs = filter(x -> any(occursin.(possible_libc_names, Ref(x))), dllist())
-    if isempty(libcs)
-        @debug("Could not find libc, so will look for musl instead")
-        libcs = filter(x -> any(occursin.(possible_musl_names, Ref(x))), dllist())
-    end
-    isempty(libcs) && error("Could not find libc or musl, unable to call uname()")
+    libc_name = @static if libc(Base.BinaryPlatforms.HostPlatform()) == "glibc"
+        "libc.so"
+    else
+        "ld-musl-x86_64.so"
+    end
+    libcs = filter(x -> occursin(libc_name, x), dllist())
+    isempty(libcs) && error("Could not find libc, unable to call uname()")
     libc = dlopen(first(libcs))
     uname_hdl = dlsym(libc, :uname)
```

Comment:
Any particular reason why we do all this dance to get the handle of `uname` instead of just doing
```julia
julia> uname_struct = zeros(UInt8, 2048);

julia> ccall(:uname, Cint, (Ptr{UInt8},), uname_struct)
0
```
?  We directly ccall symbols in libc without specifying the library in https://github.com/staticfloat/Sandbox.jl/blob/ed87f079edb1cef60c32ad5fbfbcd38c70a1314a/src/utils.jl#L192 and https://github.com/staticfloat/Sandbox.jl/blob/ed87f079edb1cef60c32ad5fbfbcd38c70a1314a/src/utils.jl#L197

Reply:
Okay, you've convinced me.  I think it's unlikely that we'll ever have a conflicting `:uname` symbol.